llvm/lib/Target/AMDGPU/AMDGPUCombine.td

```diff
@@ -125,14 +125,14 @@ def sign_extension_in_reg : GICombineRule<
   (apply [{ applyCombineSignExtendInReg(*${sign_inreg}, ${matchinfo}); }])>;
 
 // Do the following combines :
-// fmul x, select(y, A, B)   -> ldexp (x, select i32 (y, a, b))
-// fmul x, select(y, -A, -B) -> ldexp ((fneg x), select i32 (y, a, b))
-def combine_fmul_with_select_to_ldexp : GICombineRule<
+// fmul x, select(y, A, B)   -> fldexp (x, select i32 (y, a, b))
+// fmul x, select(y, -A, -B) -> fldexp ((fneg x), select i32 (y, a, b))
+def combine_fmul_with_select_to_fldexp : GICombineRule<
```

Comment:
We really should have complex patterns and constant xforms like selection patterns

Reply:
Can you kindly elaborate, as dealing with pattern matching for first time in Isel!